src/System.Private.Reflection.Execution/src/Internal/Reflection/Execution/ExecutionEnvironmentImplementation.MappingTables.cs

```diff
@@ -1089,17 +1089,45 @@ private unsafe bool TryGetMethodForOriginalLdFtnResult_Inner(NativeFormatModuleI
                         {
                             unsafe
                             {
-                                if (fieldAccessMetadata.Flags.HasFlag(FieldTableFlags.IsGcSection))
-                                    fieldAddress = *(IntPtr*)TypeLoaderEnvironment.Instance.TryGetGcStaticFieldData(declaringTypeHandle) + fieldAccessMetadata.Offset;
+                                if ((fieldAccessMetadata.Flags & FieldTableFlags.IsGcSection) != 0)
+                                {
+                                    IntPtr gcStaticsBase = *(IntPtr*)TypeLoaderEnvironment.Instance.TryGetGcStaticFieldData(declaringTypeHandle);
+#if CORERT
+                                    // The GC statics are allocated at runtime instead of being stored in the image.
+                                    // A double indirection is needed to get the the GC statics base allocated by the runtime.
+                                    gcStaticsBase = **(IntPtr**)gcStaticsBase;
```

Comment:
Is this a GC hole? This just indirected through the GC handle table and `gcStaticsBase` points to an object allocated on the heap, no?

Maybe we need a new FieldAccessor that can either hold on to the object reference in a GC compatible way, or an accessor that can do the extra indirection in a safe manner.